File: JavaScript_Courses/JavaScript_Basics/Preparation/04_For_Loop_Exercise/08_Tennis_Ranklist.js
function tennisRanklist(input) {
    let pointsW = 2000;
    let pointsF = 1200;
    let pointsSF = 720;

    let tournamentsCount = Number(input[0]);
    let startingPoints = Number(input[1]);

    let totalPoints = startingPoints;
    let tournamentsW = 0;


    for (let i = 2; i < input.length; i++) {
        let currentResult = input[i];

        if (currentResult == "W") {
            totalPoints += pointsW;
            tournamentsW++;
        } else if (currentResult == "F") {
            totalPoints += pointsF;
        } else if (currentResult == "SF") {
            totalPoints += pointsSF;
        }
    }

    console.log(`Final points: ${totalPoints}`);
    console.log(`Average points: ${Math.floor((totalPoints - startingPoints) / tournamentsCount)}`);
    console.log(`${(tournamentsW / tournamentsCount * 100).toFixed(2)}%`);
}


// tennisRanklist(["5",
//     "1400",
//     "F",
//     "SF",
//     "W",
//     "W",
//     "SF"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/04_For_Loop_Exercise/08_Tennis_Ranklist.py
import math

tournaments_count = int(input())
starting_points = int(input())

gained_points = 0
wins_count = 0
for _ in range(tournaments_count):
    current_result = input()  # 'W', 'F', 'SF'

    if current_result == 'W':
        gained_points += 2000
        wins_count += 1
    elif current_result == 'F':
        gained_points += 1200
    elif current_result == 'SF':  # else:
        gained_points += 720

final_points = starting_points + gained_points
avg_points = gained_points / tournaments_count
wins_percent = wins_count / tournaments_count * 100

print(f'Final points: {final_points}')
print(f'Average points: {math.floor(avg_points)}')
print(f'{wins_percent:.2f}%')
